winml/lib/Api/impl/TensorBase.h

```diff
@@ -826,13 +1007,14 @@ struct TensorBase : TBase {
     // Ensure that this call is being called with the correct template parameters
     ASSERT_TEMPLATE_PARAMETERS<int8_t, uint8_t>();
 
-    uint32_t size;
-    int8_t* pBuffer;
-
-    // Get the data pointer and size
-    std::tie(size, pBuffer) = GetCpuResource()->buffer();
+    // Ensure that the Set APIs are only called when there is 1 buffer.
+    // These APIs are only called when the tensor is being constructed from various collection and pointer public APIs.
+    // They should always be backed by a single underlying buffer.
+    FAIL_FAST_HR_IF(E_ILLEGAL_METHOD_CALL, GetCpuResource()->num_buffers() != 1);
 
-    std::transform(begin(data), end(data), pBuffer, [](auto element) { return static_cast<int8_t>(element); });
+    auto buffer = GetCpuResource()->buffer();
+    auto element_data = static_cast<int8_t*>(buffer.data());
+    std::transform(begin(data), end(data), element_data, [](auto element) { return static_cast<int8_t>(element); });
```

Comment:
Is it safe to assume `GetCpuResource()->buffer().size() >= data.size()`?

Reply:
the buffers are allocated based on the provided shape.
It is possible that if the data size is larger than the shape provided that this will freak out. I have not tried it... but i imagine it gets caught and transmitted over the abi correctly...
This change doesnt upate this behavior.